frontend/src/board/board.test.ts

```diff
@@ -46,3 +47,93 @@ test(
     ])
   },
 );
+
+
+test('Checking eligible moves in the North direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'North')).toStrictEqual(
+    { row: 0, column: 15 })
+}
+)
+test('Checking eligible moves in the South direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'South')).toStrictEqual(
+    { row: 15, column: 15 })
+}
+)
+
+test('Checking eligible moves in the East direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 14, 'East')).toStrictEqual(
+    {row: 6, column: 15})
+}
+)
+test('Checking eligible move in the West direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'West')).toStrictEqual(
+  {row: 6, column: 0})
+}
+)
+
+test("Should return target cell in path", () => {
+  let board = new Board();
+  if (board.cells[15] && board.cells[15][14]) {
```

Comment:
This condition should always be true - and it's alright if we crash in a unit test, that's a valuable data point

Can we just do `board.cells[15]!.[14]!.isTarget = true;`?